### PR TITLE
Added save contract with local storage functionality

### PR DESCRIPTION
**Pull request by mmnavarr**
# Title
Added save contract with local storage functionality

Screenshot:
image

Video:
https://user-images.githubusercontent.com/3385743/176285889-bc258b57-c470-4500-aef5-cadda48cad64.mov

### CODE REVIEW

File: utils/useLocalStorage.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useState } from 'react'
```
Comment (0xbe1) [inline utils/useLocalStorage.ts:1]: Can we use https://github.com/streamich/react-use/blob/master/docs/useLocalStorage.md from react-use? It is available in our dependencies

Comment (mmnavarr) [reply utils/useLocalStorage.ts:1]: I don't see why not, I can swap it in and see how it plays with the existing usage of the hook

Comment (0xbe1) [reply utils/useLocalStorage.ts:1]: awesome!

Comment (mmnavarr) [reply utils/useLocalStorage.ts:1]: Have to install `react-use` to use that hook and in doing so includes a lot of other things

Swapping out the hook causes some weird UI behavior --
image

The only code change that caused the above..
image

**Comment by 0xbe1** [comment]
Can you push to a separate branch so that I can play with it? Thanks!

**Pull request closed without merging**